### PR TITLE
Add tests for PG 15's new aggregate commands on distributed tables

### PR DESCRIPTION
**Pull request by onderkalaci**
Add tests for PG commits https://github.com/postgres/postgres/commit/400fc6b6487ddf16aa82c9d76e5cfbe64d94f660 and https://github.com/postgres/postgres/commit/7ae1619bc5b1794938c7387a766b8cae34e38d8a.

Both tests include pushdown and pull to coordinator type of agg. execution.

### CODE REVIEW

File: src/test/regress/sql/pg15.sql [hunk 1]
```diff
@@ -337,8 +337,55 @@ DELETE FROM PKTABLE WHERE id = 1 OR id = 2;
 SELECT * FROM FKTABLE ORDER BY id;
 
 
--- Clean up
 \c - - - :master_port
 
+create table reservations ( room_id integer not null, booked_during daterange );
```
Comment (hanefi) [inline src/test/regress/sql/pg15.sql:342]: It looks like we forget to set `search_path` after a `\c` command. This should be fine after a rebase as I already fixed that in #6306

File: src/test/regress/sql/pg15.sql [hunk 2]
```diff
@@ -337,8 +337,55 @@ DELETE FROM PKTABLE WHERE id = 1 OR id = 2;
...
+(6, daterange('2018-07-05', '2018-07-10')),
+-- 7: two ranges that meet: no gap or overlap
+(7, daterange('2018-07-01', '2018-07-07')),
+(7, daterange('2018-07-07', '2018-07-14')),
+-- 8: an empty range
+(8, 'empty'::daterange);
+SELECT create_distributed_table('reservations', 'room_id');
+
+-- should be fine to pushdown range_agg
+SELECT   room_id, range_agg(booked_during)
```
Comment (hanefi) [inline src/test/regress/sql/pg15.sql:369]: This test might be flaky. An [excerpt from PG docs](https://www.postgresql.org/docs/current/functions-aggregate.html) :

> The aggregate functions array_agg, json_agg, jsonb_agg, json_object_agg, jsonb_object_agg, string_agg, and xmlagg, as well as similar user-defined aggregate functions, produce meaningfully different result values depending on the order of the input values. This ordering is unspecified by default, but can be controlled by writing an ORDER BY clause within the aggregate call.

How about

```suggestion
SELECT   room_id, range_agg(booked_during ORDER BY booked_during)
```

File: src/test/regress/sql/pg15.sql [hunk 3]
```diff
@@ -337,8 +337,55 @@ DELETE FROM PKTABLE WHERE id = 1 OR id = 2;
...
+FROM     reservations
+GROUP BY room_id
+ORDER BY room_id;
+
+-- should be fine to apply range_agg on the coordinator
+SELECT   room_id + 1, range_agg(booked_during)
```
Comment (hanefi) [inline src/test/regress/sql/pg15.sql:375]: Risk of being flaky. Can use an ORDER BY

File: src/test/regress/sql/pg15.sql [hunk 4]
```diff
@@ -337,8 +337,55 @@ DELETE FROM PKTABLE WHERE id = 1 OR id = 2;
...
+create table xid8_t1 (x xid8, y int);
+insert into xid8_t1 values ('0', 1), ('010', 2), ('42', 3), ('0xffffffffffffffff', 4), ('-1', 5);
+SELECT create_distributed_table('xid8_t1', 'x');
+select min(x), max(x) from xid8_t1 ORDER BY 1,2;
+select min(x), max(x) from xid8_t1 GROUP BY x ORDER BY 1,2;
+select min(x), max(x) from xid8_t1 GROUP BY y ORDER BY 1,2;
+
+-- Clean up
 \set VERBOSITY terse
+SET client_min_messages TO ERROR;
```
Comment (hanefi) [inline src/test/regress/sql/pg15.sql:390]: I already did this. Should be easy to drop this during a rebase.

File: src/test/regress/sql/pg15.sql [hunk 5]
```diff
@@ -337,8 +337,55 @@ DELETE FROM PKTABLE WHERE id = 1 OR id = 2;
...
+GROUP BY room_id + 1
+ORDER BY room_id + 1;
+
+-- min() and max() for xid8
+create table xid8_t1 (x xid8, y int);
+insert into xid8_t1 values ('0', 1), ('010', 2), ('42', 3), ('0xffffffffffffffff', 4), ('-1', 5);
+SELECT create_distributed_table('xid8_t1', 'x');
+select min(x), max(x) from xid8_t1 ORDER BY 1,2;
+select min(x), max(x) from xid8_t1 GROUP BY x ORDER BY 1,2;
+select min(x), max(x) from xid8_t1 GROUP BY y ORDER BY 1,2;
```
Comment (hanefi) [inline src/test/regress/sql/pg15.sql:386]: Honestly, I do not see much value in adding tests for these functions as I do not see any reason why that is different than any other UDF.

However, there is no harm in adding the tests here as it does not really take much CPU time to finish those.

**Review by naisila — APPROVED** [review]
LGTM 🚀

**Pull request merged**